### PR TITLE
Feature10

### PR DESCRIPTION
**Pull request by Cags23**
Can now select multiple polygons and preform computations on the selected and visible polygons

### CODE REVIEW

File: app/static/js/main.js [hunk 1]
```diff
@@ -10,14 +11,22 @@ var polygons = {
         shape.path = e.overlay.getPath();
         shape.id = new Date().getTime() + Math.floor(Math.random() * 1000);
         this.collection[shape.id] = shape;
-        this.setSelection(shape);
+        //this.setSelection(shape);
```
Comment (alwill) [inline app/static/js/main.js]: remove commented code

File: app/static/js/main.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 var map;
 var polygons = {
+    scollec: {},
```
Comment (alwill) [inline app/static/js/main.js]: better name

File: app/static/js/main.js [hunk 3]
```diff
@@ -43,21 +52,36 @@ var polygons = {
         shape.path = poly.getPath();
         shape.id = new Date().getTime() + Math.floor(Math.random() * 1000);
         this.collection[shape.id] = shape;
-        this.setSelection(shape);
+        //this.setSelection(shape);
```
Comment (alwill) [inline app/static/js/main.js]: remove commented code

File: app/static/js/main.js [hunk 4]
```diff
@@ -66,13 +90,33 @@ var polygons = {
...
+            this.selectedShape = shape;
+        }
+        this.selectedShape.set('draggable', false);
+        this.selectedShape.set('editable', false);
+        this.selectedShape = null;
+        if(Object.keys(this.scollec).length !== null) {
+            delete this.scollec[shape.id];
+        }
+    },
+    isInSCollec: function(shape) {
```
Comment (alwill) [inline app/static/js/main.js]: better function name

File: app/static/js/main.js [hunk 5]
```diff
@@ -276,29 +321,32 @@ function addPolygonToList(polygonID, computation) {
     $("#show-hide-" + polygonID).on("click", function(e) {
         var polygon = polygons.collection[polygonID];
         showHidePolygonButton(this, polygon);
+        isHidden = !isHidden;
```
Comment (alwill) [inline app/static/js/main.js:332]: Maybe im wrong and not quite sure what is happening but wont this always be true?

Comment (alwill) [reply app/static/js/main.js:332]: Why do we need this variable?

File: app/static/js/main.js [hunk 6]
```diff
@@ -158,6 +202,7 @@ function initialize() {
             type: "POST",
             url: "/api/find_intersections",
             success: function(data) {
+                console.log(data);
```
Comment (jflowaa) [inline app/static/js/main.js]: Remove console logs

File: app/static/js/main.js (hunk 5 above)
Comment (alwill) [reply app/static/js/main.js:332]: Nvm I see it is line 333

File: app/static/js/main.js [hunk 7]
```diff
@@ -276,29 +321,32 @@ function addPolygonToList(polygonID, computation) {
...
         var polygonID = $(this).parent().attr("id");
         var polygon = polygons.collection[polygonID];
         deletePolygonButton(this, polygon);
     })
     $("#" + polygonID).on("click", function(e) {
-       clearPolygonListBorders();
        var polygon = polygons.collection[polygonID];
-       polygons.setSelection(polygon);
-       createPolygonListBorder(polygonID);
+       if (!polygons.isInSCollec(polygon) && isHidden !== true) {
```
Comment (jflowaa) [inline app/static/js/main.js]:
```
if (!polygons.isInSCollec(polygon) && !isHidden) 
```

File: app/static/js/main.js [hunk 8]
```diff
@@ -326,7 +326,7 @@ function addPolygonToList(polygonID, computation) {
     })
     $("#" + polygonID).on("click", function(e) {
        var polygon = polygons.collection[polygonID];
-       if (!polygons.isInSCollec(polygon) && isHidden !== true) {
+       if (!polygons.isInSCollec(polygon) && !isHidden) {
            polygons.multipleSelection(polygon);
            createPolygonListBorder(polygonID);
        }else {
```
Comment (jflowaa) [inline app/static/js/main.js]: It's an eye sore, you're missing a space. I think line `66` is the same.

```
} else {
```

File: app/static/js/main.js [hunk 9]
```diff
@@ -276,29 +315,32 @@ function addPolygonToList(polygonID, computation) {
...
+function createPolygonListBorder(polygonID) {
     document.getElementById(polygonID).style.border = "2px solid black";
 }
 
-function clearPolygonListBorders() {
-    for(var polygonID in polygons.collection) {
-        document.getElementById(polygonID).style.border = "none";
-    }
+function clearPolygonListBorders(polygonID) {
+    document.getElementById(polygonID).style.border = "none";
```
Comment (jflowaa) [inline app/static/js/main.js:357]: This is _fine_ however you're using JavaScript DoM to modify the HTML, it's just that we've been using jQuery to modify the HTML. They both do the same thing but in my opinion it doesn't flow with what we currently have

File: app/static/js/main.js [hunk 10]
```diff
@@ -276,29 +315,32 @@ function addPolygonToList(polygonID, computation) {
...
         var polygonID = $(this).parent().attr("id");
         var polygon = polygons.collection[polygonID];
         deletePolygonButton(this, polygon);
     })
     $("#" + polygonID).on("click", function(e) {
-       clearPolygonListBorders();
        var polygon = polygons.collection[polygonID];
-       polygons.setSelection(polygon);
-       createPolygonListBorder(polygonID);
+       if (!polygons.isInSCollec(polygon) && !isHidden) {
```
Comment (jflowaa) [inline app/static/js/main.js]: Does this still work how that it has been refactored?

```
 polygons.isInSCollec()
```

Either way it should also be renamed.

Comment (Cags23) [reply app/static/js/main.js]: Fixed

File: app/static/js/main.js [hunk 11]
```diff
@@ -50,6 +50,7 @@ var polygons = {
         shape.path = poly.getPath();
         shape.id = new Date().getTime() + Math.floor(Math.random() * 1000);
         this.collection[shape.id] = shape;
+        this.deselecteAll();
```
Comment (jflowaa) [inline app/static/js/main.js]: Typo -> `deselectAll()`

**Pull request merged**